src/bert_featurizer_solution.py

```diff
@@ -0,0 +1,119 @@
+from pathlib import Path
+from typing import Union
+
+import pandas as pd
+import torch.cuda
+from catboost import CatBoostRegressor
+from sklearn.model_selection import train_test_split
+
+from bert_featurizer import BertFeatureExtractor
+from src.constant_predictor import load_train_test_df
+from src.metrics import MSEMetric
+from src.solution import Solution
+from src.text_cleaning.spell_checker import SmartSpellChecker
+from src.text_cleaning.text_feature_extractor import \
+    HandcraftedTextFeatureExtractor
+from src.text_cleaning.text_preprocessing import TextPreprocessor
+
+
+class BertWithHandcraftedFeaturePredictor(Solution):
+    device = 'GPU' if torch.cuda.is_available() else None
+
+    def __init__(self, config: dict):
+        super(BertWithHandcraftedFeaturePredictor, self).__init__()
+        spellcheck = SmartSpellChecker()
+
+        self.feature_extractor = HandcraftedTextFeatureExtractor(spellcheck)
+        self.text_preprocessing = TextPreprocessor(spellcheck)
+        self.bert = BertFeatureExtractor(model_name=config['model_name'])
+
+        # classification model for each column
+        self.columns = ['cohesion', 'syntax', 'vocabulary', 'phraseology', 'grammar', 'conventions']
+        self.models = [
+            CatBoostRegressor(
+                iterations=config['catboost_iter'],
+                task_type=self.device,
+                verbose=False,
+            ) for _ in range(len(self.columns))
+        ]
+
+    def _preprocess_texts(self, X: pd.Series):
```

Comment:
1. Bad naming: preprocessing is smth that _changes_. Image preprocessing may be cropping, resizing, blurring. Text preprocessing may be fixing typos, removing extra spaces, etc. Building a dataframe with features is not preprocessing
2. Please add annotations for the returned value

Reply:
Yep